burn-tch/src/ops/tensor.rs

```diff
@@ -221,6 +221,20 @@ impl<E: TchElement> TensorOps<TchBackend<E>> for TchBackend<E> {
         TchOps::index_assign(tensor, indexes, value)
     }
 
+    fn mask_scatter<const D: usize>(
+        tensor: TchTensor<E, D>,
+        mask: TchTensor<bool, D>,
+        source: TchTensor<E, D>,
+    ) -> TchTensor<E, D> {
+        TchTensor::binary_ops_tensor(
+            tensor,
+            source,
+            |tensor, source| tensor.f_masked_scatter_(&mask.tensor, source).unwrap(),
```

Comment:
You may have to cast to `Kind::Bool` like with `mask_fill`, you may add a test to check if it's needed?

Reply:
It seems there's no need for casting when return value is a `TchTensor<TchElement>`, while those with `TchTensor<bool>` as return type should be converted to `tch::Kind::Bool` in clousures standing for mutable. And does it already covered in the tests across `burn-tensor` and `burn-tensor`?

I'm not sure if there's any misunderstanding:cold_sweat: